lib/tasks/publishing_api.rake

```diff
@@ -234,6 +234,18 @@ namespace :publishing_api do
       puts "Finished queuing items for Publishing API"
     end
 
+    desc "Republish all documents with draft editions"
+    task all_drafts: :environment do
+      editions = Edition.draft.includes(:document)
+
+      puts "Enqueueing #{editions.count} documents"
+      editions.find_each do |edition|
+        document_id = edition.document.id
+        PublishingApiDocumentRepublishingWorker.perform_async_in_queue("bulk_republishing", document_id, true)
```

Comment:
I wonder if we actually need to queue this in Whitehall?  Publishing API has its own queues, including a low priority queue for bulk publishing.

If we don't need to queue here, then something like `Whitehall::PublishingApi.save_draft(edition, bulk_publishing: true)` will republish only the draft edition of the document and we won't need the `.includes(:document)` on line 239 either.

Reply:
My concern with directly calling the Publishing API (i.e. `Whitehall::PublishingApi.save_draft`) is that we'll be processing tens of thousands of draft editions. If the rake tasks fails or times out mid-way though execution (e.g. if Publishing API throws an exception), we'll need to start re-publishing again from the beginning.

Processing them async with Sidekiq allows us to protect ourselves from that possibility as each edition can be retried independently.

Reply:
We've talked it through and agreed to keep the async worker approach, given the number of documents involved.

Marking this thread as resolved.